container/src/main/java/com/flipkart/poseidon/core/PoseidonServlet.java

```diff
@@ -103,7 +103,7 @@ protected void doTrace(HttpServletRequest request, HttpServletResponse response)
     protected void doRequest(HttpMethod method, HttpServletRequest httpRequest, HttpServletResponse httpResponse) throws IOException {
         setRequestContext(httpRequest);
 
-        PoseidonRequest request = new PoseidonRequest(httpRequest);
+        PoseidonRequest request = new PoseidonRequest(httpRequest, method);
```

Comment:
Not required. Next line is setting the method in a attribute of PoseidonRequest.